stdlib/public/TensorFlow/Ops.swift

```diff
@@ -1597,52 +1597,257 @@ public extension Tensor where Scalar : Numeric {
+      self.beginMask = beginMask
+      self.endMask = endMask
+      self.ellipsisMask = ellipsisMask
+      self.newAxisMask = newAxisMask
+      self.squeezeAxisMask = squeezeAxisMask
+    }
+  }
+
+  @inlinable @inline(__always)
+  @differentiable(wrt: self, vjp: _vjpSubscript)
+  subscript(_ indexPath: IndexPath) -> Tensor {
+    get {
+      return Raw.stridedSlice(
+        self, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        beginMask: indexPath.beginMask, endMask: indexPath.endMask, 
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask, 
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+    set {
+      self = Raw.tensorStridedSliceUpdate(
+        self, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        value: newValue, beginMask: indexPath.beginMask, endMask: indexPath.endMask, 
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask, 
+        shrinkAxisMask: indexPath.squeezeAxisMask)
+    }
+  }
+
+  @inlinable @inline(__always)
+  subscript(_ indices: TensorSliceIndexProtocol...) -> Tensor {
+    get {
+      return self[IndexPath(indices.map { $0.sliceIndex })]
+    }
+    set {
+      self[IndexPath(indices.map { $0.sliceIndex })] = newValue
+    }
+  }
+
+  @inlinable @inline(__always)
+  internal func _vjpSubscript(_ indexPath: IndexPath) -> (Tensor, (Tensor) -> Tensor) {
+    return (self[indexPath], { [shape = shapeTensor] v in
+      return Tensor<Scalar>._pullbackSubscript(v, indexPath, shape)
+    })
+  }
+
+  @inlinable @inline(__always)
+  @differentiable(wrt: seed, vjp: _vjpPullbackSubscript)
+  internal static func _pullbackSubscript(
+    _ seed: Tensor,
+    _ indexPath: IndexPath,
+    _ shape: Tensor<Int32>
+  ) -> Tensor {
+    return Raw.stridedSliceGrad(
+      shape: shape, begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+      dy: seed, beginMask: indexPath.beginMask, endMask: indexPath.endMask,
+      ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask,
+      shrinkAxisMask: indexPath.squeezeAxisMask)
+  }
+
+  @inlinable @inline(__always)
+  internal static func _vjpPullbackSubscript(
+    _ seed: Tensor,
+    _ indexPath: IndexPath,
+    shape: Tensor<Int32>
+  ) -> (Tensor, (Tensor) -> Tensor) {
+    return (Tensor<Scalar>._pullbackSubscript(seed, indexPath, shape), { v in
+      return v[IndexPath(
+        begin: indexPath.begin, end: indexPath.end, strides: indexPath.strides,
+        beginMask: indexPath.beginMask, endMask: indexPath.endMask,
+        ellipsisMask: indexPath.ellipsisMask, newAxisMask: indexPath.newAxisMask,
+        squeezeAxisMask: indexPath.squeezeAxisMask)]
+    })
+  }
+}
+
+public extension Tensor.IndexPath {
+  @inlinable @inline(__always)
+  init(_ indices: [TensorSliceIndex]) {
+    precondition(!indices.isEmpty, "The index path cannot be empty.")
+    precondition(indices.count(where: {
```

Comment:
```suggestion
    precondition(indices.count { $0 == .ellipsis }, ...)
```

If `==` doesn't work, you can make `TensorSliceIndex` conform to `Equatable`.